src/AutoRest.CSharp/Common/Output/Models/Types/HelperTypeProviders/ModelSerializationExtensionsProvider.cs

```diff
@@ -398,31 +403,50 @@ public MethodBodyStatement WriteNumberValue(Utf8JsonWriterExpression writer, Val
 
         private Method BuildWriteObjectValueMethod()
         {
-            var valueParameter = new Parameter("value", null, typeof(object), null, ValidationType.None, null);
-            var signature = new MethodSignature(
-                Name: _writeObjectValueMethodName,
-                Summary: null,
-                Description: null,
-                Modifiers: _methodModifiers,
-                ReturnType: null,
-                ReturnDescription: null,
-                Parameters: new[] { KnownParameters.Serializations.Utf8JsonWriter, valueParameter });
-            var value = (ValueExpression)valueParameter;
-            var writer = new Utf8JsonWriterExpression(KnownParameters.Serializations.Utf8JsonWriter);
-            var body = new SwitchStatement(value)
+            ValueExpression value;
+            Utf8JsonWriterExpression writer;
+            ParameterReference options;
+            MethodSignature signature = GetWriteObjectValueMethodSignature(null, out value, out writer, out options);
+            return new Method(signature, new MethodBodyStatement[]
+            {
+                writer.WriteObjectValue(new TypedValueExpression(typeof(object), value), options)
+            });
+        }
+
+        private Method BuildWriteObjectValueMethodGeneric()
+        {
+            ValueExpression value;
+            Utf8JsonWriterExpression writer;
+            ParameterReference options;
+            MethodSignature signature = GetWriteObjectValueMethodSignature(new[] { _t }, out value, out writer, out options);
+            List<SwitchCase> cases = new List<SwitchCase>
             {
-                // null case
                 new(Null, new MethodBodyStatement[]
                 {
                     writer.WriteNullValue(),
                     Break
-                }),
-                // serializable case
-                BuildWriteObjectValueSwitchCase(Configuration.ApiTypes.IUtf8JsonSerializableType, "serializable", serializableVariable => new MethodBodyStatement[]
-                {
-                    new InvokeInstanceMethodStatement(serializableVariable, Configuration.ApiTypes.IUtf8JsonSerializableWriteName, writer),
-                    Break
-                }),
+                })
+            };
+            if (Configuration.UseModelReaderWriter)
+            {
+                cases.Add(
+                  BuildWriteObjectValueSwitchCase(new CSharpType(typeof(IJsonModel<>), _t), "jsonModel", jsonModel => new MethodBodyStatement[]
```

Comment:
here we have a simpler expression:
```csharp
new SwitchCase(new DeclarationExpression(typeof(IJsonModel<object>), "jsonModel", out var jsonModel), ...)
```
The `DeclarationExpression` will write:
```csharp
IJsonModel<object> jsonModel
```
which defines `jsonModel` as a `VariableReference` for latter code to use.